.circleci/config.yml

```diff
@@ -67,6 +70,46 @@ jobs:
           root: /tmp/keep-client
           paths:
             - docker-images
+
+  cross_compile_client:
+    executor: docker-xgo
+    steps:
+      - setup_remote_docker:
+          docker_layer_caching: true
+      - checkout
+      - run:
+          name: Compile binaries
+          working_directory: ~/project
+          command: xgo -v -x -out keep-client --targets=linux/amd64 .
+      # xgo doesn't respect `-dest` flag to point a destination directory so we
+      # need to move the output from a place the command saved them
+      - run:
+          name: Move binaries
+          command: |
+            mkdir -p /tmp/keep-client/bin
+            mv /build/keep-client* /tmp/keep-client/bin
+      - persist_to_workspace:
+          root: /tmp/keep-client
+          paths:
+            - bin
+
+  publish_client_binaries:
+    executor: gcp-cli/default
+    steps:
+      - attach_workspace:
+          at: /tmp/keep-client
+      - gcp-cli/install
+      - gcp-cli/initialize:
+          google-project-id: GOOGLE_PROJECT_ID
+          google-compute-zone: GOOGLE_COMPUTE_ZONE_A
+          # This param doesn't actually set anything, leaving here as a reminder to check when they fix it.
+          gcloud-service-key: GCLOUD_SERVICE_KEY
+      - run:
+          name: Upload client binaries
+          command: |
+            cd /tmp/keep-client/bin
+            gsutil -m cp * gs://${CLIENT_BIN_BUCKET}/keep-core/
```

Comment:
(1) I don't think we should upload these anywhere during regular builds.
(2) I think we should attach these to GitHub releases somehow during RC builds.
(3) For non-RC builds we need to consider a good way to provide well-packaged and signed artifacts, but that means we'll need to do local builds.